src/node_port.py

```diff
@@ -154,4 +155,59 @@ def delete_unit_service(self) -> None:
             else:
                 raise
 
+    def _node_name(self, unit_name: str) -> str:
+        """Return the node name for this unit's pod ip."""
+        try:
+            pod = self.client.get(
+                Pod,
+                name=unit_name.replace("/", "-"),
+                namespace=self.namespace,
+            )
+        except ApiError as e:
+            if e.status.code == 403:
+                self.on_deployed_without_trust()
+                return
+
+        return pod.spec.nodeName
+
+    def get_node_ip(self, unit_name: str) -> Optional[str]:
+        """Return node IP for the provided unit."""
+        try:
+            node = self.client.get(
+                Node,
+                name=self._node_name(unit_name),
+                namespace=self.namespace,
+            )
+        except ApiError as e:
+            if e.status.code == 403:
+                logger.error("Could not delete service, application needs `juju trust`")
+                self.on_deployed_without_trust()
+                return
```

Comment:
Are we ignoring other errors on purpose here? or do we need a `raise` after the return?